src/PrestaShopBundle/Routing/Converter/LegacyUrlConverter.php

```diff
@@ -125,9 +125,7 @@ public function convertByRequest(Request $request)
         $this->router->getContext()->fromRequest($request);
         $this->checkAlreadyMatchingRoute($request->getRequestUri());
 
-        $parameters = array_merge($request->query->all(), $request->request->all());
-
-        return $this->convertByParameters($parameters);
+        return $this->convertByParameters($request->query->all());
```

Comment:
Yeah when the legacy url converter was introduced along with the LegacuUrlListener we used a simple redirection which lost the initial method, so the workaround was to merge all data to make sure they are not lost
Now that the listener has been improved and is based on a 308 redirection (which keeps the method and body content) this becomes useless